src/client/AutherClient.js

```diff
@@ -0,0 +1,52 @@
+export class AutherClient {
+  #location = window.location
+
+  constructor ({ redirectUri, autherUrl, http }) {
+    this.redirectUri = redirectUri
+    this.autherUrl = autherUrl
+    this.http = http
+  }
+
+  #buildOauthUrl = () => {
+    const returnUrl = new URL(this.redirectUri)
+    const redirectUrl = new URL("/login", this.autherUrl)
+
+    redirectUrl.searchParams.append("return_url", returnUrl)
+
+    return redirectUrl.toString()
+  }
+
+  login = () => {
+    return this.#location.replace(this.#buildOauthUrl())
+  }
+
+  logout = accessToken => {
+    if (!accessToken) {
+      throw new Error("invalid.access_token")
+    }
+
+    return this.http({
+      path: "/tokens/revoke",
+      headers: { Authorization: `Bearer ${accessToken}` },
+    })
+  }
+
+  getTokens = authorizationCode => {
+    if (!authorizationCode) {
+      throw new Error("invalid.authorization_code")
+    }
+
+    return this.http({
+      path: "/tokens/initiate",
```

Comment:
is it ok to hardcode "/tokens/initiate" and "/tokens/revoke" paths? maybe it's better to pass them to `init` method?